package/src/bin/omicron-package.rs

```diff
@@ -303,8 +325,63 @@ async fn get_sha256_digest(path: &PathBuf) -> Result<Digest> {
     Ok(context.finish())
 }
 
+async fn download_prebuilt(
+    progress: &PackageProgress,
+    package_name: &str,
+    repo: &str,
+    commit: &str,
+    expected_digest: &Vec<u8>,
+    path: &Path,
+) -> Result<()> {
+    progress.set_message("downloading prebuilt".into());
+    let url = format!(
+        "https://buildomat.eng.oxide.computer/public/file/oxidecomputer/{}/image/{}/{}",
+        repo,
+        commit,
+        path.file_name().unwrap().to_string_lossy(),
+    );
+    let response = reqwest::Client::new()
+        .get(&url)
+        .send()
+        .await
+        .with_context(|| format!("failed to get {url}"))?;
+    progress.set_length(
+        response
+            .content_length()
+            .ok_or_else(|| anyhow!("Missing Content Length"))?,
+    );
+    let mut file = tokio::fs::File::create(&path)
```

Comment:
I just noticed that we don't explicitly remove the old file.  Does this remove it, or at least truncate it?

Reply:
It truncates it on the second attempt. If it helps, the hash won't match on partial downloads, so it would know that it needs to retry if we re-execute.